subsys/bluetooth/mesh/shell/shell_dtt_cli.c

```diff
@@ -74,7 +80,13 @@ static int cmd_instance_get_all(const struct shell *shell, size_t argc, char *ar
 
 static int cmd_instance_set(const struct shell *shell, size_t argc, char *argv[])
 {
-	uint8_t elem_idx = (uint8_t)strtol(argv[1], NULL, 0);
+	int err = 0;
+	uint8_t elem_idx = shell_strtoul(argv[1], 10, &err);
```

Comment:
Is it possible to pass 0 as base to infer from string? If so, I'd prefer for element index to have base = 0, as I'd expect to be able to supply it as hex with 0x-prefix. Goes for all models.